### PR TITLE
[2.0] Clean up docblocks in Configuration class - #1796

### PR DESCRIPTION
**Pull request by caciobanu**
|      Q       |   A
|------------- | -----------
| Type         | cleanup
| BC Break     | no
| Fixed issues | #1796 

#### Summary

 Removed docblocks that seemed useless & added type hints where it was possible.

### CODE REVIEW

**Comment by malarzm** [comment]
@caciobanu thanks for the PR! The CS violations are now fixed on the `master` branch, can you please rebase and squash your commits?

**Comment by caciobanu** [comment]
@malarzm done.

File: lib/Doctrine/ODM/MongoDB/Configuration.php [hunk 1]
```diff
@@ -202,6 +175,9 @@ public function getProxyDir()
      * during each script execution.
      *
      * @return bool|int
+     *
+     * @todo: Check that it is safe to drop bool values as Doctrine\Common\Proxy\AbstractProxyFactory
```
Comment (malarzm) [inline lib/Doctrine/ODM/MongoDB/Configuration.php]: For 2.0 it'll be ok to drop `bool`, but this can be done in a separate PR :+1:

**Comment by malarzm** [comment]
Thank you @caciobanu!

**Pull request merged**